gptel-org.el

```diff
@@ -287,17 +287,17 @@ for inclusion into the user prompt for the gptel request."
 
 (defun gptel-org--link-standalone-p (object)
   "Check if link OBJECT is on a line by itself."
-  ;; Specify ancestor TYPES as list (#245)
-  (let ((par (org-element-lineage object '(paragraph))))
-    (and (= (gptel-org--element-begin object)
-            (save-excursion
-              (goto-char (org-element-property :contents-begin par))
-              (skip-chars-forward "\t ")
-              (point)))                 ;account for leading space
-                                        ;before object
-         (<= (- (org-element-property :contents-end par)
-                (org-element-property :end object))
-             1))))
+  (let* ((par (org-element-lineage object '(paragraph)))
```

Comment:
Can be simplified to just `(when-let* ((par (org-element-lineage object '(paragraph))))...)`, and you can remove the `(when (and begin end)...)`.  If there's a paragraph in the lineage, it is guaranteed to have a non-nil beginning and end positions.

You don't need to bind `begin` and `end` since their values will only be used once.

Reply:
I only ever write elisp for my literate config so I definitely need guidance with it, thank you!